workers/proto/kitchen_sink/kitchen_sink.proto

```diff
@@ -0,0 +1,328 @@
+syntax = "proto3";
+
+package temporal.omes.kitchen_sink;
+
+import "google/protobuf/duration.proto";
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/empty.proto";
+
+import "child_workflow.proto";
+import "activity.proto";
+
+import "temporal/api/common/v1/message.proto";
+import "temporal/api/failure/v1/message.proto";
+import "temporal/api/enums/v1/workflow.proto";
+
+// The input to the test overall. A copy of this constitutes everything that is needed to reproduce
+// the test.
+message TestInput {
+  WorkflowInput workflow_input = 1;
+  ClientSequence client_sequence = 2;
+  // Technically worker options should be known as well. We don't have any common format for that
+  // and creating one feels overkill to start with. Requiring the harness to print the config at
+  // startup seems good enough for now.
+}
+
+// All the client actions that will be taken over the course of this test
+message ClientSequence {
+  repeated ClientActionSet action_sets = 1;
+}
+
+// A set of client actions to execute. Client actions are always executed concurrently, since
+// sequential execution can be represented by a series of 1-size action sets.
+message ClientActionSet {
+  repeated ClientAction actions = 1;
+}
+
+message ClientAction {
+  oneof variant {
+    DoSignal do_signal = 1;
+    DoQuery do_query = 2;
+    DoUpdate do_update = 3;
+  }
+}
+
+message DoSignal {
+  oneof variant {
+    // An signal handler must exist named `do_actions_signal` which is responsible for executing
+    // the action set.
+    ActionSet do_actions = 1;
+    // Send an arbitrary signal
+    HandlerInvocation custom = 2;
+  }
+}
+
+message DoQuery {
+  oneof variant {
+    // A query must exist named `report_state` which returns the `WorkflowState` message. The input
+    // is pointless and only exists to allow testing of variably-sized query args.
+    temporal.api.common.v1.Payloads report_state = 1;
+    // Send an arbitrary query
+    HandlerInvocation custom = 2;
+  }
+}
+
+message DoUpdate {
+  oneof variant {
+    // An update must exist named `do_actions_update` which does the same thing the signal handler
+    // would do when given the provided action set. The handler should return the `WorkflowState`
+    // when done.
+    ActionSet do_actions = 1;
+    // This update request must be rejected by the workflow using a handler named `always_reject`
+    google.protobuf.Empty reject_me = 2;
+    // This update request must be accepted and then failed by the workflow using a handler named
+    // `always_fail`
+    temporal.api.failure.v1.Failure fail_update = 3;
```

Comment:
Why do we need this, shouldn't the updates action set result in a return or error?

Reply:
It could work that way, but that wasn't my plan. My thinking was to make the action set handler always return the state, and then be able to explore the other update conditions explicitly with these. I don't necessarily have any issue doing it that way though.

Reply:
It  is important for update testing I can run multiple actions and then return a value or continue as new